chemistry-opencmis-test-tck/src/test/java/org/apache/chemistry/opencmis/tck/tests/NavigationServices/NavigationServiceTest.java

```diff
@@ -0,0 +1,651 @@
+import org.apache.chemistry.opencmis.commons.data.CreatablePropertyTypes;
+import org.apache.chemistry.opencmis.commons.data.NewTypeSettableAttributes;
+import org.apache.chemistry.opencmis.commons.definitions.Choice;
+import org.apache.chemistry.opencmis.commons.definitions.ItemTypeDefinition;
+import org.apache.chemistry.opencmis.commons.definitions.PropertyDefinition;
+import org.apache.chemistry.opencmis.commons.enums.Cardinality;
+import org.apache.chemistry.opencmis.commons.enums.CmisVersion;
+import org.apache.chemistry.opencmis.commons.enums.PropertyType;
+import org.apache.chemistry.opencmis.commons.enums.Updatability;
+import org.apache.chemistry.opencmis.commons.exceptions.CmisBaseException;
+import org.apache.chemistry.opencmis.commons.exceptions.CmisObjectNotFoundException;
+import org.apache.chemistry.opencmis.commons.impl.dataobjects.AbstractPropertyDefinition;
+import org.apache.chemistry.opencmis.commons.impl.dataobjects.ChoiceImpl;
+import org.apache.chemistry.opencmis.commons.impl.dataobjects.ItemTypeDefinitionImpl;
+import org.apache.chemistry.opencmis.commons.impl.dataobjects.PropertyBooleanDefinitionImpl;
+import org.apache.chemistry.opencmis.commons.impl.dataobjects.PropertyDateTimeDefinitionImpl;
+import org.apache.chemistry.opencmis.commons.impl.dataobjects.PropertyDecimalDefinitionImpl;
+import org.apache.chemistry.opencmis.commons.impl.dataobjects.PropertyHtmlDefinitionImpl;
+import org.apache.chemistry.opencmis.commons.impl.dataobjects.PropertyIdDefinitionImpl;
+import org.apache.chemistry.opencmis.commons.impl.dataobjects.PropertyIntegerDefinitionImpl;
+import org.apache.chemistry.opencmis.commons.impl.dataobjects.PropertyStringDefinitionImpl;
+import org.apache.chemistry.opencmis.commons.impl.dataobjects.PropertyUriDefinitionImpl;
+import org.apache.chemistry.opencmis.tck.CmisTestResult;
+import org.apache.chemistry.opencmis.tck.impl.AbstractSessionTest;
+import org.apache.chemistry.opencmis.tck.runner.AbstractRunner;
+
+public class NavigationServiceTest extends AbstractSessionTest {
+
+	@Override
+	public void init(Map<String, String> parameters) {
+		super.init(parameters);
+		setName("NavigationService Tests");
+		setDescription("Creates a Item type with NavigationService");
+	}
+
+	@Override
+	public void run(Session session) throws Exception {
+		if (session.getRepositoryInfo().getCmisVersion() == CmisVersion.CMIS_1_0) {
+			addResult(createResult(SKIPPED, "Type mutability is not supported by CMIS 1.0. Test skipped!"));
+			return;
+		}
+
+		ObjectType parentType = session.getTypeDefinition(getItemTestTypeId());
+		if (parentType.getTypeMutability() == null
+				|| !Boolean.TRUE.equals(parentType.getTypeMutability().canCreate())) {
+			addResult(createResult(SKIPPED, "Test document type doesn't allow creating a sub-type. Test skipped!"));
+			return;
+		}
+
+		createTypeWithProperties(session, parentType);
+	}
+
+	private void createTypeWithProperties(Session session, ObjectType parentType) {
```

Comment:
Im seeing this function repeated, Can we move this a common helper?

Reply:
```suggestion
	private void createTypeWithProperties(Session session, ObjectType parentType) {
```
FunctionName is same, but we are doing different logic in each TestCase